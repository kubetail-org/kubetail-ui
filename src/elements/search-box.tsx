import { forwardRef } from 'react';
import { Search } from 'lucide-react';

import { Input } from '@/elements/input';
import { cn } from '@/lib/utils';

export type SearchBoxProps = React.ComponentProps<typeof Input> & {
  containerClassName?: string;
};

/**
 * A compact search box.
 * - Uses shadcn <Input/>
 * - Leading search icon
 */
const SearchBox = forwardRef<HTMLInputElement, SearchBoxProps>(
  ({ className, containerClassName, placeholder = '', ...props }, ref) => {
    const { disabled } = props;

    return (
      <div className={cn('relative w-full max-w-sm', containerClassName)}>
        <Search
          aria-hidden
          className={cn(
            'pointer-events-none absolute left-3 top-1/2 h-4 w-4 -translate-y-1/2 text-muted-foreground',
            disabled && 'opacity-50',
          )}
        />
        <Input ref={ref} type="search" placeholder={placeholder} className={cn('pl-9', className)} {...props} />
      </div>
    );
  },
);

SearchBox.displayName = 'SearchBox';

export { SearchBox };
