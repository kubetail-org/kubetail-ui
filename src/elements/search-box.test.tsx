import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { SearchBox } from './search-box';

describe('SearchBox', () => {
  it('renders contents properly', () => {
    const { asFragment } = render(<SearchBox placeholder="Enter text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('applies custom className', () => {
    const { getByPlaceholderText } = render(<SearchBox placeholder="Class test" className="custom-class" />);
    expect(getByPlaceholderText('Class test')).toHaveClass('custom-class');
  });

  it('accepts and displays value', () => {
    const { getByDisplayValue } = render(<SearchBox value="Hello" onChange={() => {}} />);
    expect(getByDisplayValue('Hello')).toBeInTheDocument();
  });

  it('renders as disabled', () => {
    const { getByPlaceholderText } = render(<SearchBox placeholder="Disabled test" disabled />);
    expect(getByPlaceholderText('Disabled test')).toBeDisabled();
  });

  it('sets aria-invalid when invalid', () => {
    const { getByPlaceholderText } = render(<SearchBox placeholder="Invalid test" aria-invalid />);
    expect(getByPlaceholderText('Invalid test')).toHaveAttribute('aria-invalid');
  });
});
