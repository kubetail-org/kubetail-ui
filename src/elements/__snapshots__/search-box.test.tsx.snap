// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBox > renders contents properly 1`] = `
<DocumentFragment>
  <div
    class="relative w-full max-w-sm"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-search pointer-events-none absolute left-3 top-1/2 h-4 w-4 -translate-y-1/2 text-muted-foreground"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m21 21-4.34-4.34"
      />
      <circle
        cx="11"
        cy="11"
        r="8"
      />
    </svg>
    <input
      class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive pl-9"
      data-slot="input"
      placeholder="Enter text"
      type="search"
    />
  </div>
</DocumentFragment>
`;
